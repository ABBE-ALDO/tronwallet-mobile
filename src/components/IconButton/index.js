import React from 'react'
import styled from 'styled-components'
import Ionicons from 'react-native-vector-icons/Ionicons'

const Wrapper = styled.TouchableOpacity`
  height: 36px;
  width: 36px;
  border-radius: 4px;
  background-color: #2E2F47;
  align-items: center;
  justify-content: center;
  shadow-color: #000000;
  shadow-offset: 0 2px;
  shadow-radius: 4px;
  shadow-opacity: 0.5;
`

const IconButton = ({ icon, ...props }) => (
  <Wrapper {...props}>
    <Ionicons
      name={icon}
      size={16}
      color='#9C9EB9'
    />
  </Wrapper>
)

export default IconButton
