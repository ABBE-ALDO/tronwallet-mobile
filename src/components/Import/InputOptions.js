import React, { PureComponent, Fragment } from 'react'
import { Clipboard } from 'react-native'
import { func } from 'prop-types'

import { HorizontalSpacer } from '../Utils'
import IconButton from '../IconButton'

class InputOptions extends PureComponent {
  static propTypes = {
    onPressScanner: func.isRequired,
    onPaste: func.isRequired
  }

  async _onPaste () {
    const content = await Clipboard.getString()
    this.props.onPaste(content)
  }

  render () {
    const { onPressScanner } = this.props

    return (
      <Fragment>
        <IconButton onPress={this._onPaste} icon='md-clipboard' />
        <HorizontalSpacer />
        <IconButton onPress={onPressScanner} icon='ios-qr-scanner' />
      </Fragment>
    )
  }
}

export default InputOptions
