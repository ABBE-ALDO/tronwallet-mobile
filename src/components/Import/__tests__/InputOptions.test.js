import React from 'react'
import { shallow } from 'enzyme'

import IconButton from '../../IconButton'
import InputOptions from '../InputOptions'

describe('Input options component', () => {
  const setup = propOverrides => {
    const props = Object.assign({
      onPressScanner: () => {},
      onPaste: () => {}
    }, propOverrides)

    const wrapper = shallow(
      <InputOptions {...props} />
    )

    const iconWrapper = wrapper.find(IconButton)

    return {
      props,
      wrapper,
      clipboardWrapper: iconWrapper.first(),
      qrScannerWrapper: iconWrapper.last()
    }
  }

  test('Should match when render component', () => {
    const { wrapper } = setup()
    expect(wrapper).toMatchSnapshot()
  })

  // Check way to work with Clipboard.getString
  // test('Should onPaste triggered when press clipboard icon', async () => {
  //   const onPaste = jest.fn()
  //   const { wrapper } = setup({ onPaste })

  //   const instance = wrapper.instance()
  //   await instance._onPaste()

  //   expect(onPaste).toBeCalled()
  // })

  test('Should onPressScanner triggered when press scanner icon', () => {
    const onPressScanner = jest.fn()
    const { qrScannerWrapper } = setup({ onPressScanner })

    qrScannerWrapper.simulate('press')

    expect(onPressScanner).toBeCalled()
  })
})
