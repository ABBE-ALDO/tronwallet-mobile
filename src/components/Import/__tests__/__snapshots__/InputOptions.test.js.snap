// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input options component Should match when render component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <InputOptions
    onPaste={[Function]}
    onPressScanner={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <IconButton
          bg="#2E2F47"
          highlight={false}
          icon="md-clipboard"
          iconColor="#9C9EB9"
          onPress={[Function]}
        />,
        <Styled(View)
          size="small"
        />,
        <IconButton
          bg="#2E2F47"
          highlight={false}
          icon="ios-qr-scanner"
          iconColor="#9C9EB9"
          onPress={[Function]}
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "bg": "#2E2F47",
          "highlight": false,
          "icon": "md-clipboard",
          "iconColor": "#9C9EB9",
          "onPress": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "size": "small",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "bg": "#2E2F47",
          "highlight": false,
          "icon": "ios-qr-scanner",
          "iconColor": "#9C9EB9",
          "onPress": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <IconButton
            bg="#2E2F47"
            highlight={false}
            icon="md-clipboard"
            iconColor="#9C9EB9"
            onPress={[Function]}
          />,
          <Styled(View)
            size="small"
          />,
          <IconButton
            bg="#2E2F47"
            highlight={false}
            icon="ios-qr-scanner"
            iconColor="#9C9EB9"
            onPress={[Function]}
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "bg": "#2E2F47",
            "highlight": false,
            "icon": "md-clipboard",
            "iconColor": "#9C9EB9",
            "onPress": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "size": "small",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "bg": "#2E2F47",
            "highlight": false,
            "icon": "ios-qr-scanner",
            "iconColor": "#9C9EB9",
            "onPress": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenThreeAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
